scale/README.md

```diff
@@ -0,0 +1,46 @@
+JupyterHub Kubernetes Autoscaler
+===================================
+
+### Definitions
+
+**Critical Service =** Pods whose names start with `hub, proxy, statsd`
+
+**Omitted Namespace =** Default name space, `kube-system` namespace
+
+**Omitted Service =** Pods whose names start with `cull`
+
+**Workload =**  On a node, number of pods that do not belong to `Critical Service`, do not belong to `Omitted Service`, and do not belong to `Omitted Namespace`
+
+**Capacity =** Defined number of pods that can run on a single node; the value is 0 for nodes running `Critical Service` and nodes marked `Unschedulable`
+
+**Utilization =** On certain nodes, the ratio between the sum of `Workload` and the sum of `Capacity`
+
+### Expected Behavior
+
+When `scale.py` is exected
+
+1. The autoscaler will calculate the **Utilization** of the cluster.
+2. If the **Utilization** of the cluster is between a **predefined minimum** and a **predefined maximum**, move the `Unschedulable` flag provided by Kubernetes between nodes, to make them deleted as soon as possible. Otherwise, the autoscaler will add or remove `Unschedulable` flags to approximate a **predefined optimal utilization**; if optimal utilization is not reached, new nodes can be created to meet the goal, to the predefined **maximum number of nodes**.
+**2a. Nodes running `critical service` will never be marked unschedulable**.
+3. Make sure there are at least **predefined minimum number** of nodes schedulable by removing flags or add new nodes.
+4. Shutdown all empty unschedulable nodes
+
+### How to run
+
+0. Read `settings.py` to make sure you like the current settings.
+1. Ensure a kubectl proxy **running in the context you want to scale** is listening at `http://API_HOST:API_PORT/`, both values were defined in `settings.py`. By default, it is `http://localhost:18080`. See `https://kubernetes.io/docs/user-guide/kubectl/kubectl_proxy/` for details.
```

Comment:
I'd really like to get rid of this - we should be using https://github.com/kubernetes-incubator/client-python. It will automatically use a [service account](https://kubernetes.io/docs/user-guide/service-accounts/) when running inside the cluster, or read the `.kube/config` file for authentication otherwise. 

Since this code doesn't have a large number of kube API interactions yet, this would be the easiest time to change this - changing this in the future is more difficult.